### PR TITLE
feat: Use cargo --locked whenever possible

### LINKED ISSUE #3547: ci: always run `cargo` with `--locked`
## Summary

Since https://github.com/libp2p/rust-libp2p/pull/3399 we are tracking the `Cargo.lock` file via version control. Currently CI does not fail when the checked in `Cargo.lock` is not up-to-date.

## Possible Solution

Replace all `cargo XXX` with `cargo --locked XXX`.

Can you think of a simpler solution @thomaseizinger?

## Would you like to work on fixing this bug?

Yes

### PR DESCRIPTION
**Pull request by mxinden**
## Description

> Since https://github.com/libp2p/rust-libp2p/pull/3399 we are tracking the `Cargo.lock` file via
version control. Currently CI does not fail when the checked in `Cargo.lock` is not up-to-date.

Fixes https://github.com/libp2p/rust-libp2p/issues/3547.

### CODE REVIEW

**Review by jxs — APPROVED** [review]
LGTM, TIL I wasn't aware about this feature :) Though maybe we only need it on one of the jobs like `test`.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -227,7 +227,7 @@ jobs:
       - uses: r7kamura/rust-problem-matchers@d58b70c4a13c4866d96436315da451d8106f8f08 #v1.3.0
 
       - name: Check formatting
-        run: cargo fmt -- --check
+        run: cargo --locked fmt -- --check
```
Comment (thomaseizinger) [inline .github/workflows/ci.yml:230]: This seems unnecessary?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -270,7 +270,7 @@ jobs:
 
       - uses: Swatinem/rust-cache@6fd3edff6979b79f87531400ad694fb7f2c84b1f # v2.2.1
 
-      - run: cargo install --version 0.10.0 pb-rs --locked
+      - run: cargo --locked install --version 0.10.0 pb-rs --locked
```
Comment (thomaseizinger) [inline .github/workflows/ci.yml:273]: Here too. This doesn't actually use our lock file.

**Review by thomaseizinger — COMMENTED** [review]
It seems a bit verbose to add the flag everywhere. How about we just add it to the clippy job?

**Comment by thomaseizinger** [comment]
Check https://github.com/libp2p/rust-libp2p/pull/3587 for an alternative to this.

**Comment by thomaseizinger** [comment]
Closing as it has been superseded by https://github.com/libp2p/rust-libp2p/pull/3587 and https://github.com/libp2p/rust-libp2p/pull/3589.

**Pull request closed without merging**